apps/application-system/api/src/app/modules/application/lifecycle/application-lifecycle.service.ts

```diff
@@ -40,19 +43,12 @@ export class ApplicationLifeCycleService {
     return this.processingApplications
   }
 
-  private async getJustOneById(id: string) {
-    const application = await this.applicationService.findOneById(id)
-    if (application) {
-      this.processingApplications.push({
-        pruned: false,
-        application,
-        failedAttachments: {},
-      })
-    }
-  }
-
   private async fetchApplicationsToBePruned() {
-    const applications = await this.applicationService.findAllDueToBePruned()
+    const applications = (await this.applicationService.findAllDueToBePruned()) as Pick<
+      Application,
+      'id' | 'attachments' | 'answers' | 'externalData'
+    >[]
```

Comment:
Why the type cast? You should only have `id` and `attachments` here, same as the function returns.

Reply:
Well later on update it will be populated with the resulting updated values . Checking these values in the tests.

Reply:
I see. Not great to lose type safety just for the tests. Wonder if there's a better way, but whatever :)